### ISSUE TITLE
CA2014 triggered erroneously for stackalloc'd collection

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#5693]
### Analyzer

**Diagnostic ID**: [CA2014](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2014): `Do not use stackalloc in loops`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

A stackalloc inside the body of a loop is a problem and can lead to stackoverflow.  But a stackalloc as the collection being enumerated itself is fine.

### Steps To Reproduce

```C#
class C
{
    public static void M()
    {
        foreach (char c in stackalloc char[] { 'a', 'b', 'c' })
        {
        }
    }
}
```

### Expected behavior

No warning.

### Actual behavior

CA2014 triggers.
